### PR TITLE
add history list view to the REST api tab

### PR DESCRIPTION
**Pull request by pariskol**
I fixed the default border color of code area when not focused.

### CODE REVIEW

**Comment by reugn** [comment]
Hey @pariskol. Already fixed and released v0.4.0.

**Comment by pariskol** [comment]
Hi Eugene!  I added a new feature to rest api tester tab. I added a listview which stores requests history with capabilities of importing, exporting history to file via context menu options.

File: src/main/java/com/github/reugn/devtools/controllers/RestAPIController.java [hunk 1]
```diff
@@ -1,20 +1,144 @@
...
+        	}
+        });
+        
+        MenuItem clearHistoryItem = new MenuItem("Clear History");
+        clearHistoryItem.setOnAction(event -> {
+        	historyListView.getItems().clear();
+        	RestService.REQ_HISTORY_LIST.clear();
+        });
+        
+        MenuItem openRequestItem = new MenuItem("Open request");
```
Comment (reugn) [inline src/main/java/com/github/reugn/devtools/controllers/RestAPIController.java]: Add "Delete request" option.

File: src/main/java/com/github/reugn/devtools/services/RestService.java [hunk 2]
```diff
@@ -1,54 +1,103 @@
...
+		long t2 = System.currentTimeMillis() - t1;
 
-        String responseHeaders = getResponseHeaders(conn.getHeaderFields());
-        conn.disconnect();
-        return new RestResponse(status, responseBody, responseHeaders, t2);
+		String responseHeaders = getResponseHeaders(conn.getHeaderFields());
+		conn.disconnect();
+
+		Request req = new Request(uri, requestMethod, headers, body);
+		REQ_HISTORY_LIST.add(req);
```
Comment (reugn) [inline src/main/java/com/github/reugn/devtools/services/RestService.java]: Don't add, if the exact request already exists in the list and move it to the bottom as a last used.

File: src/main/java/com/github/reugn/devtools/controllers/RestAPITabController.java [hunk 3]
```diff
@@ -119,9 +173,36 @@ private void addHeader() {
...
+			for (String key : req.getHeaders().keySet()) {
+				HBox n = FXMLLoader.load(this.getClass().getResource("/views/rest_api_header.fxml"));
+				HttpHeadersTextField htf = (HttpHeadersTextField) n.getChildren().get(0);
+				htf.setText(key);
+				TextField tf = (TextField) n.getChildren().get(1);
+				tf.setText(req.getHeaders().get(key));
+
+				requestHeadersVBox.getChildren()
+						.add(n);
+			}
```
Comment (reugn) [inline src/main/java/com/github/reugn/devtools/controllers/RestAPITabController.java]:
```java
boolean first = true;
for (Map.Entry<String, String> entry : req.getHeaders().entrySet()) {
    if (first) {
        HBox hBox = (HBox) requestHeadersVBox.getChildren().get(0);
        setHeader(hBox, entry.getKey(), entry.getValue());
        first = false;
    } else {
        HBox hBox = FXMLLoader.load(getClass().getResource("/views/rest_api_header.fxml"));
        setHeader(hBox, entry.getKey(), entry.getValue());
        requestHeadersVBox.getChildren().add(hBox);
    }
}
```

File: src/main/java/com/github/reugn/devtools/controllers/RestAPITabController.java [hunk 4]
```diff
@@ -119,9 +173,36 @@ private void addHeader() {
...
+			methodComboBox.getItems().forEach(x -> {
+				if (x.equals(req.getMethod()))
+					methodComboBox.getSelectionModel().select(x);
+			});
+		} catch (IOException e) {
+			log.warn(e.getMessage(), e);
+		}
+	}
```
Comment (reugn) [inline src/main/java/com/github/reugn/devtools/controllers/RestAPITabController.java:210]:
```java
private void setHeader(HBox hBox, String key, String value) {
    HttpHeadersTextField htf = (HttpHeadersTextField) hBox.getChildren().get(0);
    htf.setText(key);
    TextField tf = (TextField) hBox.getChildren().get(1);
    tf.setText(value);
}
```

**Review by reugn — CHANGES_REQUESTED** [review]
Hi @pariskol, this is great! Please see my review below.

File: src/main/java/com/github/reugn/devtools/controllers/RestAPIController.java (hunk 1 above)
Comment (pariskol) [reply src/main/java/com/github/reugn/devtools/controllers/RestAPIController.java]: Done

File: src/main/java/com/github/reugn/devtools/services/RestService.java (hunk 2 above)
Comment (pariskol) [reply src/main/java/com/github/reugn/devtools/services/RestService.java]: Done

File: src/main/java/com/github/reugn/devtools/controllers/RestAPITabController.java (hunk 3 above)
Comment (pariskol) [reply src/main/java/com/github/reugn/devtools/controllers/RestAPITabController.java]: Done

File: src/main/java/com/github/reugn/devtools/controllers/RestAPITabController.java (hunk 4 above)
Comment (pariskol) [reply src/main/java/com/github/reugn/devtools/controllers/RestAPITabController.java:210]: Done

**Comment by pariskol** [comment]
Check please!

**Pull request merged**